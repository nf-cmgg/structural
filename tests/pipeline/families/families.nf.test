nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"

    tag "pipeline"
    tag "pipeline_families"

    options "-stub"

    test("one family of 3 samples - no concat") {

        when {
            params {
                outdir = "${outputDir}"
                input = "${baseDir}/tests/pipeline/families/samplesheet_family3.csv"
                callers = "delly,smoove,manta,expansionhunter,qdnaseq,wisecondorx"
                max_cpus   = 2
                max_memory = '6.GB'
                max_time   = '6.h'
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.stdout,
                    path("$outputDir")
                        .list()
                        .findAll {
                            !it.toString().endsWith("pipeline_info")
                        }
                        .collect {
                            it.list().collect { it.toString().split("/")[-1] }
                        },
                    path("$outputDir").list().collect{
                        it.toString().split("/")[-1]
                    }
                ).match("family with 3 samples - no concat")}
            )
        }

    }

    test("one family of 3 samples - concat") {

        when {
            params {
                outdir = "${outputDir}"
                input = "${baseDir}/tests/pipeline/families/samplesheet_family3.csv"
                callers = "delly,smoove,manta,expansionhunter,qdnaseq,wisecondorx"
                max_cpus   = 2
                max_memory = '6.GB'
                max_time   = '6.h'
                concat_output = true
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.stdout,
                    path("$outputDir")
                        .list()
                        .findAll {
                            !it.toString().endsWith("pipeline_info")
                        }
                        .collect {
                            it.list().collect { it.toString().split("/")[-1] }
                        },
                    path("$outputDir").list().collect{
                        it.toString().split("/")[-1]
                    }
                ).match("family with 3 samples - concat")}
            )
        }

    }

    test("one family of 2 samples + 1 sample - no concat") {

        when {
            params {
                outdir = "${outputDir}"
                input = "${baseDir}/tests/pipeline/families/samplesheet_family2_sample1.csv"
                callers = "delly,smoove,manta,expansionhunter,qdnaseq,wisecondorx"
                max_cpus   = 2
                max_memory = '6.GB'
                max_time   = '6.h'
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.stdout,
                    path("$outputDir")
                        .list()
                        .findAll {
                            !it.toString().endsWith("pipeline_info")
                        }
                        .collect {
                            it.list().collect { it.toString().split("/")[-1] }
                        },
                    path("$outputDir").list().collect{
                        it.toString().split("/")[-1]
                    }
                ).match("family with 2 samples + 1 sample - no concat")}
            )
        }

    }

    test("one family of 2 samples + 1 sample - concat") {

        when {
            params {
                outdir = "${outputDir}"
                input = "${baseDir}/tests/pipeline/families/samplesheet_family2_sample1.csv"
                callers = "delly,smoove,manta,expansionhunter,qdnaseq,wisecondorx"
                max_cpus   = 2
                max_memory = '6.GB'
                max_time   = '6.h'
                concat_output = true
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.stdout,
                    path("$outputDir")
                        .list()
                        .findAll {
                            !it.toString().endsWith("pipeline_info")
                        }
                        .collect {
                            it.list().collect { it.toString().split("/")[-1] }
                        },
                    path("$outputDir").list().collect{
                        it.toString().split("/")[-1]
                    }
                ).match("family with 2 samples + 1 sample - concat")}
            )
        }

    }

}
