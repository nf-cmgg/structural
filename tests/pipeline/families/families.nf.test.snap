{
    "family with 2 samples + 1 sample - concat": {
        "content": [
            [
                
            ],
            [
                [
                    "PosCon1.qdnaseq.abberations.bed",
                    "PosCon1.qdnaseq.bed",
                    "PosCon1.qdnaseq.cna",
                    "PosCon1.qdnaseq_segments.txt",
                    "PosCon1.vcf.gz",
                    "PosCon1.vcf.gz.tbi",
                    "PosCon1.wisecondorx_aberrations.bed",
                    "PosCon1.wisecondorx_bins.bed",
                    "PosCon1.wisecondorx_chr_statistics.txt",
                    "PosCon1.wisecondorx_segments.bed",
                    "chr1.png",
                    "chr22.png",
                    "genome_wide.png",
                    "statistics.out"
                ],
                [
                    "PosCon2.qdnaseq.abberations.bed",
                    "PosCon2.qdnaseq.bed",
                    "PosCon2.qdnaseq.cna",
                    "PosCon2.qdnaseq_segments.txt",
                    "PosCon2.vcf.gz",
                    "PosCon2.vcf.gz.tbi",
                    "PosCon2.wisecondorx_aberrations.bed",
                    "PosCon2.wisecondorx_bins.bed",
                    "PosCon2.wisecondorx_chr_statistics.txt",
                    "PosCon2.wisecondorx_segments.bed",
                    "chr1.png",
                    "chr22.png",
                    "genome_wide.png",
                    "statistics.out"
                ],
                [
                    "PosCon3.qdnaseq.abberations.bed",
                    "PosCon3.qdnaseq.bed",
                    "PosCon3.qdnaseq.cna",
                    "PosCon3.qdnaseq_segments.txt",
                    "PosCon3.vcf.gz",
                    "PosCon3.vcf.gz.tbi",
                    "PosCon3.wisecondorx_aberrations.bed",
                    "PosCon3.wisecondorx_bins.bed",
                    "PosCon3.wisecondorx_chr_statistics.txt",
                    "PosCon3.wisecondorx_segments.bed",
                    "chr1.png",
                    "chr22.png",
                    "genome_wide.png",
                    "statistics.out"
                ],
                [
                    "multiqc_data",
                    "multiqc_plots",
                    "multiqc_report.html"
                ],
                [
                    "test_family.vcf.gz",
                    "test_family.vcf.gz.tbi"
                ]
            ],
            [
                "PosCon1",
                "PosCon2",
                "PosCon3",
                "multiqc",
                "pipeline_info",
                "test_family"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.02.0"
        },
        "timestamp": "2024-03-13T12:12:35.02840081"
    },
    "family with 2 samples + 1 sample - no concat": {
        "content": [
            [
                
            ],
            [
                [
                    "PosCon1.cnv.vcf.gz",
                    "PosCon1.cnv.vcf.gz.tbi",
                    "PosCon1.expansionhunter.vcf.gz",
                    "PosCon1.expansionhunter.vcf.gz.tbi",
                    "PosCon1.qdnaseq.abberations.bed",
                    "PosCon1.qdnaseq.bed",
                    "PosCon1.qdnaseq.cna",
                    "PosCon1.qdnaseq_segments.txt",
                    "PosCon1.sv.vcf.gz",
                    "PosCon1.sv.vcf.gz.tbi",
                    "PosCon1.wisecondorx_aberrations.bed",
                    "PosCon1.wisecondorx_bins.bed",
                    "PosCon1.wisecondorx_chr_statistics.txt",
                    "PosCon1.wisecondorx_segments.bed",
                    "chr1.png",
                    "chr22.png",
                    "genome_wide.png",
                    "statistics.out"
                ],
                [
                    "PosCon2.cnv.vcf.gz",
                    "PosCon2.cnv.vcf.gz.tbi",
                    "PosCon2.expansionhunter.vcf.gz",
                    "PosCon2.expansionhunter.vcf.gz.tbi",
                    "PosCon2.qdnaseq.abberations.bed",
                    "PosCon2.qdnaseq.bed",
                    "PosCon2.qdnaseq.cna",
                    "PosCon2.qdnaseq_segments.txt",
                    "PosCon2.sv.vcf.gz",
                    "PosCon2.sv.vcf.gz.tbi",
                    "PosCon2.wisecondorx_aberrations.bed",
                    "PosCon2.wisecondorx_bins.bed",
                    "PosCon2.wisecondorx_chr_statistics.txt",
                    "PosCon2.wisecondorx_segments.bed",
                    "chr1.png",
                    "chr22.png",
                    "genome_wide.png",
                    "statistics.out"
                ],
                [
                    "PosCon3.cnv.vcf.gz",
                    "PosCon3.cnv.vcf.gz.tbi",
                    "PosCon3.expansionhunter.vcf.gz",
                    "PosCon3.expansionhunter.vcf.gz.tbi",
                    "PosCon3.qdnaseq.abberations.bed",
                    "PosCon3.qdnaseq.bed",
                    "PosCon3.qdnaseq.cna",
                    "PosCon3.qdnaseq_segments.txt",
                    "PosCon3.sv.vcf.gz",
                    "PosCon3.sv.vcf.gz.tbi",
                    "PosCon3.wisecondorx_aberrations.bed",
                    "PosCon3.wisecondorx_bins.bed",
                    "PosCon3.wisecondorx_chr_statistics.txt",
                    "PosCon3.wisecondorx_segments.bed",
                    "chr1.png",
                    "chr22.png",
                    "genome_wide.png",
                    "statistics.out"
                ],
                [
                    "multiqc_data",
                    "multiqc_plots",
                    "multiqc_report.html"
                ],
                [
                    "test_family.cnv.vcf.gz",
                    "test_family.cnv.vcf.gz.tbi",
                    "test_family.repeats.vcf.gz",
                    "test_family.repeats.vcf.gz.tbi",
                    "test_family.sv.vcf.gz",
                    "test_family.sv.vcf.gz.tbi"
                ]
            ],
            [
                "PosCon1",
                "PosCon2",
                "PosCon3",
                "multiqc",
                "pipeline_info",
                "test_family"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.02.0"
        },
        "timestamp": "2024-03-29T09:51:16.761133108"
    },
    "family with 3 samples - no concat": {
        "content": [
            [
                
            ],
            [
                [
                    "PosCon1.cnv.vcf.gz",
                    "PosCon1.cnv.vcf.gz.tbi",
                    "PosCon1.expansionhunter.vcf.gz",
                    "PosCon1.expansionhunter.vcf.gz.tbi",
                    "PosCon1.qdnaseq.abberations.bed",
                    "PosCon1.qdnaseq.bed",
                    "PosCon1.qdnaseq.cna",
                    "PosCon1.qdnaseq_segments.txt",
                    "PosCon1.sv.vcf.gz",
                    "PosCon1.sv.vcf.gz.tbi",
                    "PosCon1.wisecondorx_aberrations.bed",
                    "PosCon1.wisecondorx_bins.bed",
                    "PosCon1.wisecondorx_chr_statistics.txt",
                    "PosCon1.wisecondorx_segments.bed",
                    "chr1.png",
                    "chr22.png",
                    "genome_wide.png",
                    "statistics.out"
                ],
                [
                    "PosCon2.cnv.vcf.gz",
                    "PosCon2.cnv.vcf.gz.tbi",
                    "PosCon2.expansionhunter.vcf.gz",
                    "PosCon2.expansionhunter.vcf.gz.tbi",
                    "PosCon2.qdnaseq.abberations.bed",
                    "PosCon2.qdnaseq.bed",
                    "PosCon2.qdnaseq.cna",
                    "PosCon2.qdnaseq_segments.txt",
                    "PosCon2.sv.vcf.gz",
                    "PosCon2.sv.vcf.gz.tbi",
                    "PosCon2.wisecondorx_aberrations.bed",
                    "PosCon2.wisecondorx_bins.bed",
                    "PosCon2.wisecondorx_chr_statistics.txt",
                    "PosCon2.wisecondorx_segments.bed",
                    "chr1.png",
                    "chr22.png",
                    "genome_wide.png",
                    "statistics.out"
                ],
                [
                    "PosCon3.cnv.vcf.gz",
                    "PosCon3.cnv.vcf.gz.tbi",
                    "PosCon3.expansionhunter.vcf.gz",
                    "PosCon3.expansionhunter.vcf.gz.tbi",
                    "PosCon3.qdnaseq.abberations.bed",
                    "PosCon3.qdnaseq.bed",
                    "PosCon3.qdnaseq.cna",
                    "PosCon3.qdnaseq_segments.txt",
                    "PosCon3.sv.vcf.gz",
                    "PosCon3.sv.vcf.gz.tbi",
                    "PosCon3.wisecondorx_aberrations.bed",
                    "PosCon3.wisecondorx_bins.bed",
                    "PosCon3.wisecondorx_chr_statistics.txt",
                    "PosCon3.wisecondorx_segments.bed",
                    "chr1.png",
                    "chr22.png",
                    "genome_wide.png",
                    "statistics.out"
                ],
                [
                    "multiqc_data",
                    "multiqc_plots",
                    "multiqc_report.html"
                ],
                [
                    "test_family.cnv.vcf.gz",
                    "test_family.cnv.vcf.gz.tbi",
                    "test_family.repeats.vcf.gz",
                    "test_family.repeats.vcf.gz.tbi",
                    "test_family.sv.vcf.gz",
                    "test_family.sv.vcf.gz.tbi"
                ]
            ],
            [
                "PosCon1",
                "PosCon2",
                "PosCon3",
                "multiqc",
                "pipeline_info",
                "test_family"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.02.0"
        },
        "timestamp": "2024-03-29T09:41:10.690402797"
    },
    "family with 3 samples - concat": {
        "content": [
            [
                
            ],
            [
                [
                    "PosCon1.qdnaseq.abberations.bed",
                    "PosCon1.qdnaseq.bed",
                    "PosCon1.qdnaseq.cna",
                    "PosCon1.qdnaseq_segments.txt",
                    "PosCon1.vcf.gz",
                    "PosCon1.vcf.gz.tbi",
                    "PosCon1.wisecondorx_aberrations.bed",
                    "PosCon1.wisecondorx_bins.bed",
                    "PosCon1.wisecondorx_chr_statistics.txt",
                    "PosCon1.wisecondorx_segments.bed",
                    "chr1.png",
                    "chr22.png",
                    "genome_wide.png",
                    "statistics.out"
                ],
                [
                    "PosCon2.qdnaseq.abberations.bed",
                    "PosCon2.qdnaseq.bed",
                    "PosCon2.qdnaseq.cna",
                    "PosCon2.qdnaseq_segments.txt",
                    "PosCon2.vcf.gz",
                    "PosCon2.vcf.gz.tbi",
                    "PosCon2.wisecondorx_aberrations.bed",
                    "PosCon2.wisecondorx_bins.bed",
                    "PosCon2.wisecondorx_chr_statistics.txt",
                    "PosCon2.wisecondorx_segments.bed",
                    "chr1.png",
                    "chr22.png",
                    "genome_wide.png",
                    "statistics.out"
                ],
                [
                    "PosCon3.qdnaseq.abberations.bed",
                    "PosCon3.qdnaseq.bed",
                    "PosCon3.qdnaseq.cna",
                    "PosCon3.qdnaseq_segments.txt",
                    "PosCon3.vcf.gz",
                    "PosCon3.vcf.gz.tbi",
                    "PosCon3.wisecondorx_aberrations.bed",
                    "PosCon3.wisecondorx_bins.bed",
                    "PosCon3.wisecondorx_chr_statistics.txt",
                    "PosCon3.wisecondorx_segments.bed",
                    "chr1.png",
                    "chr22.png",
                    "genome_wide.png",
                    "statistics.out"
                ],
                [
                    "multiqc_data",
                    "multiqc_plots",
                    "multiqc_report.html"
                ],
                [
                    "test_family.vcf.gz",
                    "test_family.vcf.gz.tbi"
                ]
            ],
            [
                "PosCon1",
                "PosCon2",
                "PosCon3",
                "multiqc",
                "pipeline_info",
                "test_family"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.02.0"
        },
        "timestamp": "2024-03-13T12:02:09.935078056"
    }
}